src/attributes/decorator.js

```diff
@@ -34,6 +41,8 @@ function attributesDecorator(schema, schemaOptions = {}) {
 
     schema = Schema.normalize(schema, schemaOptions);
 
+    schema[SCHEMA_OPTIONS] = schemaOptions;
```

Comment:
We should avoid adding stuff to the schema like that, prefer to use `Object.defineProperty`  (check the example on line 46) and set this attribute to be non-enumerable. Also, I believe this should go inside `Schema.normalize`.